### PR TITLE
380 mapi sandbox

### PR DESCRIPTION
**Pull request by Orta21**
Ref: 380

### Dependencies

- Upstream: Document stuff (i dont think there is a PR yet)

### Description

Stub the commonwell payloads

### Release Plan

- [ ] Upstream dependencies are met
- [ ] Once approved merge

### CODE REVIEW

File: api/app/src/external/commonwell/link/create.ts
Comment (Orta21) [inline api/app/src/external/commonwell/link/create.ts:39]: This was not necessary since its in the reset function

File: api/app/src/external/commonwell/patient.ts
Comment (Orta21) [inline api/app/src/external/commonwell/patient.ts:89]: So this function calls many other functions that contain commonwell actions. I was stubbing each one of those responses but then realized that this function is really only a wrapper for commonwell and doesnt interact with our system except for the line below which i think is fine but let me know what you guys think.

`await storeIds({ commonwellPatientId, personId });`

File: packages/packages/commonwell-sdk/src/models/patient.ts
Comment (Orta21) [inline packages/packages/commonwell-sdk/src/models/patient.ts:16]: On a patient this link is actual just a regular link not a network link

File: api/app/src/external/commonwell/link/create.ts
Comment (Goncharo) [inline api/app/src/external/commonwell/link/create.ts:39]: General comment:

This all looks good, however creating a mock CW object could be cleaner. With this approach, we need to modify the logic flow everywhere we talk to CW -> this could be more error prone, and more difficult to maintain, since we'll always have to think about sandbox when developing.

What if instead we change the logic in  `makeCommonWellAPI()` to return a CommonWellMock object if the API is running in Sandbox? This way, we don't have to change the flow of logic in the API at all, and can just mock the returns of the CW API.

Ideally, we'd refactor this (and actually we should probably make this a pattern) to have `CommonWell()` be an interface, not an actual object, and have classes that implement the interface to either be the actual API, or the mock. This way, we can just always return `CommonWell()` from `makeCommonWellAPI()`, and won't have to return 2 types, ie `CommonWell | CommonWellMock`.

Lmk what you think - happy to chat about this

**Review by Goncharo — APPROVED** [review]
Code looks fine, but I'd consider the comment I made regarding the design of this

File: api/app/src/external/commonwell/link/create.ts
Comment (leite08) [reply api/app/src/external/commonwell/link/create.ts:39]: I thought this was here so we could remove the `personId` from our local DB to match the fact the patient is not linked to that person anymore.

**Review by leite08 — COMMENTED** [review]
Will wait for the definition on the chat on slack to review this, as that might change this PR considerably.

File: api/app/src/external/commonwell/patient.ts
Comment (Goncharo) [reply api/app/src/external/commonwell/patient.ts:89]: Yeah that makes sense - note my other comment regarding the design of this

Comment (leite08) [reply api/app/src/external/commonwell/patient.ts:89]: I think it would be good to execute that logic too, which aligns with the approach we're taking forward of mocking the API - so the main logic of the app will be the same, just the API behavior changing.

File: api/app/src/external/commonwell/mock.ts
Comment (Orta21) [inline api/app/src/external/commonwell/mock.ts:282]: Once doc stuff gets merged i can add this

**Review by Goncharo — APPROVED** [review]
Niiiiiiice 🚢 🚢 🚢

File: packages/packages/commonwell-sdk/src/models/commonwell.ts
Comment (leite08) [inline packages/packages/commonwell-sdk/src/models/commonwell.ts:11]: Not a huge deal, but I suggest we rename this to `CommonWellAPI` (it matches the `makeCommonWellAPI` and it conveys what it means), rename the file to the same name, and move it to `../client` since its related to the client and not the models.

File: packages/packages/commonwell-sdk/src/models/commonwell.ts
Comment (leite08) [inline packages/packages/commonwell-sdk/src/models/commonwell.ts:12]: I think this is only used internally, so in theory no need to be exposed in the interface.

**Review by leite08 — COMMENTED** [review]
Love the approach, good job! 🙌

File: packages/packages/commonwell-sdk/src/models/commonwell.ts
Comment (Goncharo) [reply packages/packages/commonwell-sdk/src/models/commonwell.ts]: Good call - was going to comment something similar in my previous review, but the comment got lost some how 🤷

File: api/app/src/routes/medical/link.ts
Comment (Goncharo) [inline api/app/src/routes/medical/link.ts:72]: nit: as per the [tsdoc convention](https://tsdoc.org/pages/tags/returns/), we can just have do the following:

` * @returns The patient's current and potential links.`

**Review by Goncharo — APPROVED** [review]
small nit - otherwise 🚢 🚢 🚢

File: api/app/src/routes/medical/link.ts
Comment (leite08) [reply api/app/src/routes/medical/link.ts]: I was thinking about this yesterday... this makes sense for functions, but I wonder if we should have the type for endpoints. Reasoning being, doc generation likely won't be able to tell (easily) what the result type is.

Thoughts?

Comment (Goncharo) [reply api/app/src/routes/medical/link.ts]: Note that depending on how we do doc gen, these comments could be auto-generated as part of the API interface. AFAIK, there is no good way for us to do doc/SDK gen for what we need strictly off of the tsdoc.

As such, for now we could just be consistent with our own comment style (adhering to tsdoc)?

**Pull request merged**
